{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Scope of the notebook"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "I use pybiomart to retrieve the associated gene for each of the variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pybiomart\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Load the variants and Ensembl dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>variant</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>rs61751443</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>rs28935168</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>rs61751449</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>rs63750264</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>rs11136000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      variant\n",
       "0  rs61751443\n",
       "1  rs28935168\n",
       "2  rs61751449\n",
       "3  rs63750264\n",
       "4  rs11136000"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#read txt file\n",
    "variant_list = pd.read_csv('variant_list.txt',  header=0)\n",
    "variant_list.columns = ['variant']  \n",
    "variant_list['variant']=variant_list['variant'].astype(str)\n",
    "variant_list.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>display_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>ENSEMBL_MART_ENSEMBL</td>\n",
       "      <td>Ensembl Genes 98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>ENSEMBL_MART_MOUSE</td>\n",
       "      <td>Mouse strains 98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>ENSEMBL_MART_SEQUENCE</td>\n",
       "      <td>Sequence</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>ENSEMBL_MART_ONTOLOGY</td>\n",
       "      <td>Ontology</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>ENSEMBL_MART_GENOMIC</td>\n",
       "      <td>Genomic features 98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>ENSEMBL_MART_SNP</td>\n",
       "      <td>Ensembl Variation 98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>ENSEMBL_MART_FUNCGEN</td>\n",
       "      <td>Ensembl Regulation 98</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    name           display_name\n",
       "0   ENSEMBL_MART_ENSEMBL       Ensembl Genes 98\n",
       "1     ENSEMBL_MART_MOUSE       Mouse strains 98\n",
       "2  ENSEMBL_MART_SEQUENCE               Sequence\n",
       "3  ENSEMBL_MART_ONTOLOGY               Ontology\n",
       "4   ENSEMBL_MART_GENOMIC    Genomic features 98\n",
       "5       ENSEMBL_MART_SNP   Ensembl Variation 98\n",
       "6   ENSEMBL_MART_FUNCGEN  Ensembl Regulation 98"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Select the database\n",
    "server = pybiomart.Server(host='http://www.ensembl.org')\n",
    "server.list_marts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>display_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>cfamiliaris_snp</td>\n",
       "      <td>Dog Short Variants (SNPs and indels excluding ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>drerio_structvar</td>\n",
       "      <td>Zebrafish Structural Variants (GRCz11)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>ecaballus_structvar</td>\n",
       "      <td>Horse Structural Variants (EquCab3.0)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>ptroglodytes_snp</td>\n",
       "      <td>Chimpanzee Short Variants (SNPs and indels exc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>cfamiliaris_structvar</td>\n",
       "      <td>Dog Structural Variants (CanFam3.1)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>mgallopavo_snp</td>\n",
       "      <td>Turkey Short Variants (SNPs and indels excludi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>mmusculus_structvar</td>\n",
       "      <td>Mouse Structural Variants (GRCm38.p6)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>nleucogenys_snp</td>\n",
       "      <td>Gibbon Short Variants (SNPs and indels excludi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>fcatus_snp</td>\n",
       "      <td>Cat Short Variants (SNPs and indels excluding ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>hsapiens_structvar</td>\n",
       "      <td>Human Structural Variants (GRCh38.p13)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>hsapiens_snp</td>\n",
       "      <td>Human Short Variants (SNPs and indels excludin...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>sscrofa_structvar</td>\n",
       "      <td>Pig Structural Variants (Sscrofa11.1)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>oaries_snp</td>\n",
       "      <td>Sheep Short Variants (SNPs and indels excludin...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>hsapiens_snp_som</td>\n",
       "      <td>Human Somatic Short Variants (SNPs and indels ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>rnorvegicus_snp</td>\n",
       "      <td>Rat Short Variants (SNPs and indels excluding ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>btaurus_structvar</td>\n",
       "      <td>Cow Structural Variants (ARS-UCD1.2)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>mmusculus_snp</td>\n",
       "      <td>Mouse Short Variants (SNPs and indels excludin...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>chircus_snp</td>\n",
       "      <td>Goat Short Variants (SNPs and indels excluding...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>ggallus_snp</td>\n",
       "      <td>Chicken Short Variants (SNPs and indels exclud...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>ecaballus_snp</td>\n",
       "      <td>Horse Short Variants (SNPs and indels excludin...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>drerio_snp</td>\n",
       "      <td>Zebrafish Short Variants (SNPs and indels excl...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>tnigroviridis_snp</td>\n",
       "      <td>Tetraodon Short Variants (SNPs and indels excl...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>hsapiens_structvar_som</td>\n",
       "      <td>Human Somatic Structural Variants (GRCh38.p13)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>pabelii_snp</td>\n",
       "      <td>Orangutan Short Variants (SNPs and indels excl...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>scerevisiae_snp</td>\n",
       "      <td>Saccharomyces cerevisiae Short Variants (SNPs ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>oaries_structvar</td>\n",
       "      <td>Sheep Structural Variants (Oar_v3.1)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>oanatinus_snp</td>\n",
       "      <td>Platypus Short Variants (SNPs and indels exclu...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>sscrofa_snp</td>\n",
       "      <td>Pig Short Variants (SNPs and indels excluding ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>28</td>\n",
       "      <td>tguttata_snp</td>\n",
       "      <td>Zebra Finch Short Variants (SNPs and indels ex...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>29</td>\n",
       "      <td>mmulatta_structvar</td>\n",
       "      <td>Macaque Structural Variants (Mmul_10)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>30</td>\n",
       "      <td>mdomestica_snp</td>\n",
       "      <td>Opossum Short Variants (SNPs and indels exclud...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>31</td>\n",
       "      <td>mmulatta_snp</td>\n",
       "      <td>Macaque Short Variants (SNPs and indels exclud...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>32</td>\n",
       "      <td>btaurus_snp</td>\n",
       "      <td>Cow Short Variants (SNPs and indels excluding ...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                      name                                       display_name\n",
       "0          cfamiliaris_snp  Dog Short Variants (SNPs and indels excluding ...\n",
       "1         drerio_structvar             Zebrafish Structural Variants (GRCz11)\n",
       "2      ecaballus_structvar              Horse Structural Variants (EquCab3.0)\n",
       "3         ptroglodytes_snp  Chimpanzee Short Variants (SNPs and indels exc...\n",
       "4    cfamiliaris_structvar                Dog Structural Variants (CanFam3.1)\n",
       "5           mgallopavo_snp  Turkey Short Variants (SNPs and indels excludi...\n",
       "6      mmusculus_structvar              Mouse Structural Variants (GRCm38.p6)\n",
       "7          nleucogenys_snp  Gibbon Short Variants (SNPs and indels excludi...\n",
       "8               fcatus_snp  Cat Short Variants (SNPs and indels excluding ...\n",
       "9       hsapiens_structvar             Human Structural Variants (GRCh38.p13)\n",
       "10            hsapiens_snp  Human Short Variants (SNPs and indels excludin...\n",
       "11       sscrofa_structvar              Pig Structural Variants (Sscrofa11.1)\n",
       "12              oaries_snp  Sheep Short Variants (SNPs and indels excludin...\n",
       "13        hsapiens_snp_som  Human Somatic Short Variants (SNPs and indels ...\n",
       "14         rnorvegicus_snp  Rat Short Variants (SNPs and indels excluding ...\n",
       "15       btaurus_structvar               Cow Structural Variants (ARS-UCD1.2)\n",
       "16           mmusculus_snp  Mouse Short Variants (SNPs and indels excludin...\n",
       "17             chircus_snp  Goat Short Variants (SNPs and indels excluding...\n",
       "18             ggallus_snp  Chicken Short Variants (SNPs and indels exclud...\n",
       "19           ecaballus_snp  Horse Short Variants (SNPs and indels excludin...\n",
       "20              drerio_snp  Zebrafish Short Variants (SNPs and indels excl...\n",
       "21       tnigroviridis_snp  Tetraodon Short Variants (SNPs and indels excl...\n",
       "22  hsapiens_structvar_som     Human Somatic Structural Variants (GRCh38.p13)\n",
       "23             pabelii_snp  Orangutan Short Variants (SNPs and indels excl...\n",
       "24         scerevisiae_snp  Saccharomyces cerevisiae Short Variants (SNPs ...\n",
       "25        oaries_structvar               Sheep Structural Variants (Oar_v3.1)\n",
       "26           oanatinus_snp  Platypus Short Variants (SNPs and indels exclu...\n",
       "27             sscrofa_snp  Pig Short Variants (SNPs and indels excluding ...\n",
       "28            tguttata_snp  Zebra Finch Short Variants (SNPs and indels ex...\n",
       "29      mmulatta_structvar              Macaque Structural Variants (Mmul_10)\n",
       "30          mdomestica_snp  Opossum Short Variants (SNPs and indels exclud...\n",
       "31            mmulatta_snp  Macaque Short Variants (SNPs and indels exclud...\n",
       "32             btaurus_snp  Cow Short Variants (SNPs and indels excluding ..."
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Select the dataset\n",
    "mart = server['ENSEMBL_MART_SNP']\n",
    "mart.list_datasets()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "10    Human Short Variants (SNPs and indels excludin...\n",
       "Name: display_name, dtype: object"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mart.list_datasets()[mart.list_datasets()['name']=='hsapiens_snp']['display_name']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    " dataset = mart['hsapiens_snp']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>display_name</th>\n",
       "      <th>description</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>study_type</td>\n",
       "      <td>Study type</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>study_external_ref</td>\n",
       "      <td>Study External Reference</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>study_description</td>\n",
       "      <td>Study Description</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>source_name</td>\n",
       "      <td>Source name</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>associated_gene</td>\n",
       "      <td>Associated gene with phenotype</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>phenotype_name</td>\n",
       "      <td>Phenotype name</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>phenotype_description</td>\n",
       "      <td>Phenotype description</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>associated_variant_risk_allele</td>\n",
       "      <td>Associated variant risk allele</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>28</td>\n",
       "      <td>p_value</td>\n",
       "      <td>P value</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>29</td>\n",
       "      <td>set_name</td>\n",
       "      <td>Variant Set Name</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                              name                    display_name description\n",
       "20                      study_type                      Study type            \n",
       "21              study_external_ref        Study External Reference            \n",
       "22               study_description               Study Description            \n",
       "23                     source_name                     Source name            \n",
       "24                 associated_gene  Associated gene with phenotype            \n",
       "25                  phenotype_name                  Phenotype name            \n",
       "26           phenotype_description           Phenotype description            \n",
       "27  associated_variant_risk_allele  Associated variant risk allele            \n",
       "28                         p_value                         P value            \n",
       "29                        set_name                Variant Set Name            "
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Attributes\n",
    "dataset.attributes\n",
    "dataset.list_attributes()[20:30]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>display_name</th>\n",
       "      <th>description</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>associated_gene</td>\n",
       "      <td>Associated gene with phenotype</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               name                    display_name description\n",
       "24  associated_gene  Associated gene with phenotype            "
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Attributes for the gene\n",
    "attr=pd.DataFrame(dataset.list_attributes())\n",
    "attr[attr['display_name'].str.contains(\"gene\")]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>type</th>\n",
       "      <th>description</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>link_so_mini_closure</td>\n",
       "      <td>list</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>link_so_regulation_closure</td>\n",
       "      <td>list</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>link_so_motif_closure</td>\n",
       "      <td>list</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr_name</td>\n",
       "      <td>text</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>start</td>\n",
       "      <td>text</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>end</td>\n",
       "      <td>text</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>band_start</td>\n",
       "      <td>drop_down_basic_filter</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>band_end</td>\n",
       "      <td>drop_down_basic_filter</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>marker_start</td>\n",
       "      <td>drop_down_basic_filter</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>marker_end</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chromosomal_region</td>\n",
       "      <td>text</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>strand</td>\n",
       "      <td>text</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>variation_source</td>\n",
       "      <td>list</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>snp_filter</td>\n",
       "      <td>list</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>variation_synonym_source</td>\n",
       "      <td>list</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>snp_synonym_filter</td>\n",
       "      <td>list</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>study_name</td>\n",
       "      <td>list</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>study_type</td>\n",
       "      <td>list</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>phenotype_description</td>\n",
       "      <td>list</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>phenotype_source</td>\n",
       "      <td>list</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>pub_short_ref</td>\n",
       "      <td>text</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>variation_set_name</td>\n",
       "      <td>list</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>sift_prediction</td>\n",
       "      <td>list</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>sift_score</td>\n",
       "      <td>text</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>polyphen_prediction</td>\n",
       "      <td>list</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>polyphen_score</td>\n",
       "      <td>text</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>minor_allele_freq</td>\n",
       "      <td>text</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>minor_allele_freq_second</td>\n",
       "      <td>text</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>clinical_significance</td>\n",
       "      <td>list</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>with_validated</td>\n",
       "      <td>boolean</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>with_variation_citation</td>\n",
       "      <td>boolean</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>distance_to_transcript</td>\n",
       "      <td>text</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>ensembl_gene</td>\n",
       "      <td>list</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>so_consequence_name</td>\n",
       "      <td>list</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>feature_stable_id</td>\n",
       "      <td>list</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>regulatory_so_consequence_name</td>\n",
       "      <td>list</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>motif_name</td>\n",
       "      <td>list</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>motif_so_consequence_name</td>\n",
       "      <td>list</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                              name                    type description\n",
       "0             link_so_mini_closure                    list            \n",
       "1       link_so_regulation_closure                    list            \n",
       "2            link_so_motif_closure                    list            \n",
       "3                         chr_name                    text            \n",
       "4                            start                    text            \n",
       "5                              end                    text            \n",
       "6                       band_start  drop_down_basic_filter            \n",
       "7                         band_end  drop_down_basic_filter            \n",
       "8                     marker_start  drop_down_basic_filter            \n",
       "9                       marker_end                                    \n",
       "10              chromosomal_region                    text            \n",
       "11                          strand                    text            \n",
       "12                variation_source                    list            \n",
       "13                      snp_filter                    list            \n",
       "14        variation_synonym_source                    list            \n",
       "15              snp_synonym_filter                    list            \n",
       "16                      study_name                    list            \n",
       "17                      study_type                    list            \n",
       "18           phenotype_description                    list            \n",
       "19                phenotype_source                    list            \n",
       "20                   pub_short_ref                    text            \n",
       "21              variation_set_name                    list            \n",
       "22                 sift_prediction                    list            \n",
       "23                      sift_score                    text            \n",
       "24             polyphen_prediction                    list            \n",
       "25                  polyphen_score                    text            \n",
       "26               minor_allele_freq                    text            \n",
       "27        minor_allele_freq_second                    text            \n",
       "28           clinical_significance                    list            \n",
       "29                  with_validated                 boolean            \n",
       "30         with_variation_citation                 boolean            \n",
       "31          distance_to_transcript                    text            \n",
       "32                    ensembl_gene                    list            \n",
       "33             so_consequence_name                    list            \n",
       "34               feature_stable_id                    list            \n",
       "35  regulatory_so_consequence_name                    list            \n",
       "36                      motif_name                    list            \n",
       "37       motif_so_consequence_name                    list            "
      ]
     },
     "execution_count": 136,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Select filters\n",
    "dataset.filters\n",
    "dataset.list_filters()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>type</th>\n",
       "      <th>description</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>snp_filter</td>\n",
       "      <td>list</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>snp_synonym_filter</td>\n",
       "      <td>list</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  name  type description\n",
       "13          snp_filter  list            \n",
       "15  snp_synonym_filter  list            "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "filters=pd.DataFrame( dataset.list_filters())\n",
    "filters[filters['name'].str.contains(\"snp\")]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get the selected attributes through a query where the condition is the variant name\n",
    "\n",
    "#r=dataset.query(attributes=['refsnp_id','chr_name','ensembl_gene_stable_id', 'refsnp_source','chrom_start','chrom_end','p_value','minor_allele_freq','distance_to_transcript','set_name','associated_gene'], filters={'snp_filter':variant_list['variant'].values.tolist()})\n",
    "result=dataset.query(attributes=['refsnp_id','chr_name','ensembl_gene_stable_id', 'refsnp_source','chrom_start','chrom_end','associated_gene'], filters={'snp_filter':variant_list['variant'].values.tolist()})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant name</th>\n",
       "      <th>Chromosome/scaffold name</th>\n",
       "      <th>Gene stable ID</th>\n",
       "      <th>Variant source</th>\n",
       "      <th>Chromosome/scaffold position start (bp)</th>\n",
       "      <th>Chromosome/scaffold position end (bp)</th>\n",
       "      <th>Associated gene with phenotype</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>rs11136000</td>\n",
       "      <td>8</td>\n",
       "      <td>ENSG00000120885</td>\n",
       "      <td>dbSNP</td>\n",
       "      <td>27607002</td>\n",
       "      <td>27607002</td>\n",
       "      <td>CLU</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>rs139237860</td>\n",
       "      <td>14</td>\n",
       "      <td>ENSG00000176165</td>\n",
       "      <td>dbSNP</td>\n",
       "      <td>28768377</td>\n",
       "      <td>28768377</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>rs141088742</td>\n",
       "      <td>14</td>\n",
       "      <td>ENSG00000176165</td>\n",
       "      <td>dbSNP</td>\n",
       "      <td>28767873</td>\n",
       "      <td>28767873</td>\n",
       "      <td>FOXG1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>rs143223844</td>\n",
       "      <td>14</td>\n",
       "      <td>ENSG00000176165</td>\n",
       "      <td>dbSNP</td>\n",
       "      <td>28768437</td>\n",
       "      <td>28768437</td>\n",
       "      <td>FOXG1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>rs147154860</td>\n",
       "      <td>14</td>\n",
       "      <td>ENSG00000176165</td>\n",
       "      <td>dbSNP</td>\n",
       "      <td>28768440</td>\n",
       "      <td>28768440</td>\n",
       "      <td>FOXG1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Variant name Chromosome/scaffold name   Gene stable ID Variant source  \\\n",
       "0   rs11136000                        8  ENSG00000120885          dbSNP   \n",
       "1  rs139237860                       14  ENSG00000176165          dbSNP   \n",
       "2  rs141088742                       14  ENSG00000176165          dbSNP   \n",
       "3  rs143223844                       14  ENSG00000176165          dbSNP   \n",
       "4  rs147154860                       14  ENSG00000176165          dbSNP   \n",
       "\n",
       "   Chromosome/scaffold position start (bp)  \\\n",
       "0                                 27607002   \n",
       "1                                 28768377   \n",
       "2                                 28767873   \n",
       "3                                 28768437   \n",
       "4                                 28768440   \n",
       "\n",
       "   Chromosome/scaffold position end (bp) Associated gene with phenotype  \n",
       "0                               27607002                            CLU  \n",
       "1                               28768377                            NaN  \n",
       "2                               28767873                          FOXG1  \n",
       "3                               28768437                          FOXG1  \n",
       "4                               28768440                          FOXG1  "
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "result.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "writer = pd.ExcelWriter('gene_associated_SNP.xlsx')\n",
    "result.to_excel(writer,\"gene_list.xlsx\")\n",
    "writer.save()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
